src/cpp/cpp_typecheck_bases.cpp

```diff
@@ -67,6 +70,8 @@ void cpp_typecheckt::typecheck_compound_bases(struct_typet &type)
       throw 0;
     }
 
+    base_symbol_expr.type().id(ID_symbol_type);
```

Comment:
That seems redundant? I don't think there is any way we could get here with `base_symbol_expr.type().id() != ID_symbol_type`.

Reply:
Indeed, turned into invariant.